src/ceph-volume/ceph_volume/util/device.py

```diff
@@ -350,12 +350,18 @@ def is_ceph_disk_member(self):
 
     @property
     def has_bluestore_label(self):
-        out, err, ret = process.call([
-            'ceph-bluestore-tool', 'show-label',
-            '--dev', self.abspath], verbose_on_failure=False)
-        if ret:
-            return False
-        return True
+        fd = None
+        isBluestore = False
+        bluestoreDiskSignature = 'bluestore block device' # 22 bytes long
+
+        # throws OSError on failure
+        with os.open(self.abspath, os.O_RDONLY, 0) as fd:
+            # read first 22 bytes looking for bluestore disk signature
+            signature = os.read(fd, 22)
```

Comment:
the with statement doesn't seem to work with `os.open()`

Any reason for not using the builtin `open()` ?

```python
with open(self.abspath, "rb") as fd:
    signature = fd.read(22)
```

Otherwise you'll need to remove the with statement

```python
fd = os.open(self.abspath, os.O_RDONLY, 0)
signature = os.read(fd, 22)
os.close(fd)
```